### DISCUSSION TITLE
Add more metric in kyuubi server

### DISCUSSION THREAD

**Discussion by zhenjiaguo** [apache/kyuubi#1671] [Ideas]
I want to add some metric in kyuubi server. This is listed in detail as follows:

1. Fetch rows rate. Use [Meters](https://metrics.dropwizard.io/4.2.0/manual/core.html#meters) to record the rows rate that fetch results and fetch logs.
2. `BackendService` method avg call times and rate. Replace [Histograms](https://metrics.dropwizard.io/4.2.0/manual/core.html#Histograms) with [Timer](https://metrics.dropwizard.io/4.2.0/manual/core.html#Timer) to record it. Timer includes both [Meters](https://metrics.dropwizard.io/4.2.0/manual/core.html#meters) and [Histograms](https://metrics.dropwizard.io/4.2.0/manual/core.html#Histograms).
3. Added metrics for all kyuubi server operations state change rate, include INITIALIZED, PENDING, RUNNING, etc. Use [Meters](https://metrics.dropwizard.io/4.2.0/manual/core.html#meters) to record it.
4. Improve the codahale metric registered in `MetricsSystem`. Add prefix to make metric more clear. Add `ClassLoadingGaugeSet` to help us more clear about JVM classloader info.
5. `KyuubiSyncThriftClient` wait locks thread count.
6. `KyuubiSyncThriftClient` each method call time metric. Use [Timer](https://metrics.dropwizard.io/4.2.0/manual/core.html#timers) to record it.
7. Active Kyuubi Server and Engine number. Implement by check the nodes registered on zookeeper and will use [gauges](https://metrics.dropwizard.io/4.2.0/manual/core.html#gauges) to record it.

If there is any question, please let me know.

**Comment by zhenjiaguo** [comment]
@yaooqinn @ulysses-you

**Comment by yaooqinn** [comment]
SGTM
